onnxruntime/core/optimizer/gemm_sum_fusion.cc

```diff
@@ -0,0 +1,161 @@
+  ORT_ENFORCE(new_gemm_output_defs.size() == 1);
+
+  Node& new_gemm_node = graph.AddNode(graph.GenerateNodeName(gemm_node.Name() + "_sum_transformed"),
+                                      gemm_node.OpType(),
+                                      "Fused Gemm with Sum",
+                                      new_gemm_input_defs,
+                                      new_gemm_output_defs,
+                                      {},
+                                      gemm_node.Domain());
+  new_gemm_node.AddAttribute("transA", static_cast<int64_t>(transA));
+  new_gemm_node.AddAttribute("transB", static_cast<int64_t>(transB));
+  new_gemm_node.AddAttribute("alpha", static_cast<float>(alpha));
+  new_gemm_node.AddAttribute("beta", static_cast<float>(beta));
+
+  // Move both input edges from original gemm to new gemm.
+  for (auto input_edge : graph_utils::GraphEdge::GetNodeInputEdges(gemm_node)) {
+    ORT_ENFORCE(input_edge.src_arg_index < 2);
+    graph.AddEdge(input_edge.src_node, new_gemm_node.Index(), input_edge.src_arg_index, input_edge.dst_arg_index);
+    graph.RemoveEdge(input_edge.src_node, input_edge.dst_node, input_edge.src_arg_index, input_edge.dst_arg_index);
+  }
+
+  // Move all output edges from sum to new gemm.
+  for (auto output_edge : graph_utils::GraphEdge::GetNodeOutputEdges(sum_node)) {
+    ORT_ENFORCE(output_edge.src_arg_index == 0);
+    graph.AddEdge(new_gemm_node.Index(), output_edge.dst_node, output_edge.src_arg_index, output_edge.dst_arg_index);
+    graph.RemoveEdge(output_edge.src_node, output_edge.dst_node, output_edge.src_arg_index, output_edge.dst_arg_index);
+  }
+
+  // Finally, move the other sum input edge to "C" for the new gemm node.
+  // If The other sum input def is a a graph input, there is no edge to move.
+  bool sum_input_moved = false;
+  for (auto sum_input : graph_utils::GraphEdge::GetNodeInputEdges(sum_node)) {
+    if (sum_input.src_node != gemm_node.Index()) {
+      ORT_ENFORCE(!sum_input_moved);
+      graph.AddEdge(sum_input.src_node, new_gemm_node.Index(), sum_input.src_arg_index, 2);
+      graph.RemoveEdge(sum_input.src_node, sum_input.dst_node, sum_input.src_arg_index, sum_input.dst_arg_index);
+      sum_input_moved = true;
+    }
+  }
+  // Old gemm node output is no longer needed. It was previously fed into the
+  // sum node which is now also handled by the new gemm. Remove this output edge
+  // to allow the node to be removed from the graph.
+  graph_utils::RemoveNodeOutputEdges(graph, gemm_node);
+
+  ORT_ENFORCE(graph.RemoveNode(gemm_node.Index()));
+  ORT_ENFORCE(graph.RemoveNode(sum_node.Index()));
+
+  modified = RewriteRuleEffect::kRemovedCurrentNode;
+
+  return Status::OK();
+}
+
+bool GemmSumFusion::SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger&) const {
+  // Perform a series of checks. If any fail, fusion may not be performed.
+
+  // C must be missing for this fusion pattern to be valid.
```

Comment:
```suggestion
  // Tensor C (in GemmTransposeFusion's document) must be missing for this fusion pattern to be valid.
```

Reply:
C in this comment is actually the third input to the original gemm from the definition of gemm inputs, and not related to the ascii diagram. I will clarify that this "C" is related to the definition of a gemm op, and not to the diagram.